packages/api-gateway/src/services/purposeService.ts

```diff
@@ -86,7 +86,7 @@ const retrieveActiveDelegationByEServiceId = async (
     },
   });
 
-  assertOnlyOneActiveDelegationForEserviceExists(result, eserviceId);
+  assertOnlyOneActiveProducerDelegationForEserviceExists(result, eserviceId);
```

Comment:
Given this validator, you should probably remove the `kind` parameter from this function and hardcode it to `delegationKind.delegatedProducer`